scripts/migrate-tags.md

```diff
@@ -0,0 +1,158 @@
+...
+```
+
+- The above output shows two `7.0.0` entries. If any duplicate release commits are found, the script will create and push tags only on the most recent commit.
+- The user has the option to supply a custom regex pattern under `-p` to narrow down the search results for the release commits.
+
+### 3. Package will be Renamed on the first Post-Migration Release
+
+- If the migration target package will be renamed after the migration, **specify the latest release version** in `-v`.
+
+```shell
+> ./scripts/migrate-tags.sh json-rpc-middleware-stream -v 5.0.1
+```
+
+```output
+38c007a3 json-rpc-middleware-stream@5.0.1
+c34b1704 json-rpc-middleware-stream@5.0.0
+8c6b70e5 json-rpc-middleware-stream@4.2.3
+f7290013 json-rpc-middleware-stream@4.2.2
+e08455ca json-rpc-middleware-stream@4.2.1
+d90fe43d json-rpc-middleware-stream@4.2.0
+...
+```
+
+### 4. Non-Dry Mode
+
+- To override dry run mode and actually create/push tags, run the script with a `--non-dry-run` flag at the end.
+- **WARNING**: If the `-r` `--remote` option isn't specified, the tags will be pushed to the `origin` repo.
+
+```shell
+> ./scripts/migrate-tags.sh json-rpc-middleware-stream -v 5.0.1 --no-dry-run
+```
+
+```output
+Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
+To https://github.com/[USERNAME]/[FORKNAME]
+ * [new tag]           json-rpc-middleware-stream@5.0.1 -> json-rpc-middleware-stream@5.0.1
+Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
+To https://github.com/[USERNAME]/[FORKNAME]
+ * [new tag]           json-rpc-middleware-stream@5.0.0 -> json-rpc-middleware-stream@5.0.0
+Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
+
+...
+
+To https://github.com/[USERNAME]/[FORKNAME]
+ * [new tag]           json-rpc-middleware-stream@2.0.0 -> json-rpc-middleware-stream@2.0.0
+Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
+To https://github.com/[USERNAME]/[FORKNAME]
+ * [new tag]           json-rpc-middleware-stream@1.0.1 -> json-rpc-middleware-stream@1.0.1
+```
+
+## D. Verify
+
+- Check whether the tags have correctly been pushed to the remote repo.
+
+```shell
+> git ls-remote --tags origin | grep 'json-rpc-engine'
+```
+
+```output
+22ff65e0f76710188b527bd5d3f81dd2103c5514        refs/tags/@metamask/json-rpc-engine@7.0.0
+7639432339e60767a8239d681911375833bc3839        refs/tags/@metamask/json-rpc-engine@7.1.0
+23aa8d9e59d9275c0725cb0264057e082034dae9        refs/tags/@metamask/json-rpc-engine@7.1.1
+67c7fee5141f6c0bb2f459c1cb3062c02bbf6a15        refs/tags/@metamask/json-rpc-engine@7.2.0
+304f6efa4d1be2460c9d0bec48224cefcf7fd208        refs/tags/json-rpc-engine@1.0.0
+4909d7fd95a555a7ae18cb1f9840db4fe1f3c85d        refs/tags/json-rpc-engine@2.0.0
+93e2b7224f7370468466e2e5e29a2c10da016b11        refs/tags/json-rpc-engine@2.1.0
+286c2716a7b856b95f74d64edd9e653728dd031c        refs/tags/json-rpc-engine@2.2.0
+...
+```
+
+## E. Troubleshooting
+
+**WARNING**: DO NOT run this script on the core repo until you have tested the results on a fork.
```

Comment:
Nit: You can use the [alert syntax in GitHub Flavored Markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) to make this stand out:

``` suggestion
> [!WARNING]
> DO NOT run this script on the core repo until you have tested the results on a fork.